src/v/cloud_storage/topic_manifest.h

```diff
@@ -34,7 +34,7 @@ class topic_manifest final : public base_manifest {
     /// Serialize manifest object
     ///
     /// \return asynchronous input_stream with the serialized json
-    serialized_json_stream serialize() const override;
+    ss::future<serialized_json_stream> serialize() const override;
```

Comment:
nit: The topic and tx manifest don't strictly need to become async, right? Perhaps `base_manifest` could expose two methods for serialisation (one sync and one async). I don't feel strongly about it though,